ext/parquet/decoder.go

```diff
@@ -1,85 +1,47 @@
 package parquet
 
 import (
-	"encoding/binary"
-	"fmt"
 	"io"
-	"reflect"
-	"time"
 
-	kpq "github.com/bsm/parquet-go/parquet"
+	goparquet "github.com/fraugster/parquet-go"
+	"github.com/fraugster/parquet-go/floor"
 )
 
 type decoder struct {
-	cols    []*columnReader
+	pfr     *goparquet.FileReader
+	ffr     *floor.Reader
 	closers []io.Closer
 }
 
-func newDecoder(rs io.ReadSeeker, names []string, batchSize int) (*decoder, error) {
-	file, err := kpq.FileFromReader(rs)
+func newDecoder(rs io.ReadSeeker) (*decoder, error) {
```

Comment:
We can probably continue to pass names in. goparquet.NewFileReader does accept a list of names.